lib/jsonapi/utils/underscore.ex

```diff
@@ -0,0 +1,33 @@
+defmodule JSONAPI.Utils.Underscore do
+  @moduledoc """
+  Helpers for replacing underscores with dashes.
+  """
+
+  def underscore?, do: Application.get_env(:jsonapi, :underscore_to_dash, false)
+
+  def underscore(value) when is_atom(value) do
+    value
+    |> to_string
+    |> underscore
+  end
+
+  def underscore(value) when is_bitstring(value) do
+    value
+    |> String.replace("_", "-")
+    |> String.to_atom
```

Comment:
And Actually I don't think we need to "to_atom" they all turn to strings in the end.